### PR TITLE
Update YAMLSyntax.rst

### PR DESCRIPTION
**Pull request by ghost**
Updated to include a bit more depth in explaining the YAML, along with a link the Wikipedia page on YAML syntax.

### CODE REVIEW

File: docsite/rst/YAMLSyntax.rst [hunk 1]
```diff
@@ -59,6 +77,17 @@ Ansible doesn't really use these too much, but you can also specify a boolean va
     likes_emacs: TRUE
     uses_cvs: false
 
+Values can span multiple lines using *|* or *>* to include newlines or ignore them::
```
Comment (jimi-c) [inline docsite/rst/YAMLSyntax.rst]: This should include a brief description on the difference between the two syntaxes.

File: docsite/rst/YAMLSyntax.rst [hunk 2]
```diff
@@ -42,6 +42,24 @@ A dictionary is represented in a simple ``key: value`` form (the colon must be f
         job: Developer
         skill: Elite
 
+More complicated data structures are possible, such as lists of dictionaries, or dictionaries whose values are lists.  Or a mix of both::
```
Comment (docschick) [inline docsite/rst/YAMLSyntax.rst]: Editing suggestion for this sentence:  More complicated data structures are possible, such as lists of dictionaries, dictionaries whose values are lists, or a mix of both::

File: docsite/rst/YAMLSyntax.rst [hunk 3]
```diff
@@ -59,6 +77,20 @@ Ansible doesn't really use these too much, but you can also specify a boolean va
     likes_emacs: TRUE
     uses_cvs: false
 
+Values can span multiple lines using *|* or *>*.  Spanning multiple lines using a *|* will include the newlines.  Using a *>* will ignore newlines; it's used to make what would otherwise be a very long line easier to read and edit.
```
Comment (docschick) [inline docsite/rst/YAMLSyntax.rst]: instead of marking up like: _|_ or _>_
maybe: `|` and `>` would be better to use to make them stand out a bit more

**Comment by jimi-c** [comment]
Merged, as the suggestions from @docschick were incorporated.

Thanks!

**Pull request merged**